Assets/Content/Items/Consumables/Food/Consumable.cs

```diff
@@ -0,0 +1,200 @@
+    
+    // Does this food leaves trash behind? Like a chocolate bar dropping it's casing
+    public GameObject trashPrefab;
+    public bool destroyObject;
+    
+    // This handles the eating sounds
+    public AudioSource audio;
+    public AudioClip[] sounds;
+
+    // Verb used to display the interaction
+    public string consumeVerb = "eat";
+    
+    // Eat action, the target is there for when you make someone eat that food
+    // For now there's no nutriment or digestion
+    // TODO: Digestion
+    public void ConsumeAction(GameObject origin, GameObject target = null)
+    {
+        if (!audio.isPlaying && content.CurrentVolume > 0)
+        {
+            // To avoid more ifs and elses
+            if (target == null) target = origin;
+            
+            // Gets player's audio source
+            audio = target.GetComponentInChildren<AudioSource>();
+            
+            // Randomly select the audio clip and the pitch
+            audio.pitch = Random.Range(0.7f,1.5f);
+            audio.PlayOneShot(sounds[Random.Range(0, sounds.Length)]);
+            
+            Item itemInHand = origin.GetComponentInChildren<Hands>().GetItemInHand();
+            
+            // Does the portion is in the last one? Does it leaves trash? 
+            if (content.CurrentVolume == 1 && trashPrefab != null )
+            {
+                Item trash = Instantiate(trashPrefab, transform.position, transform.rotation).GetComponent<Item>();
+                
+                NetworkServer.Spawn(trash.gameObject);
+                trash.GenerateNewIcon();
+
+                if (itemInHand == null)
+                {
+                    ItemHelpers.DestroyItem(this);
+                }
+                else
+                {
+                    ItemHelpers.ReplaceItem(itemInHand, trash);
+                }
+            }
+            // Last one? Should it destroy the object?
+            if (content.CurrentVolume - contentPerUse <= 0 && destroyObject)
+            {
+                ItemHelpers.DestroyItem(this);
+            }
+            // Not last one?
+            else
+            {
+                content.RemoveMoles(contentPerUse);
+                Debug.Log(target.name + " consumed " + transform.name);
+            }
+        }
+    }
+    
+    public bool CanConsume()
+    {
+        if (audio == null)
+        {
+            audio = GetComponent<AudioSource>();
+        }
+        if (!audio.isPlaying && content.CurrentVolume > 0)
+        {
+            return true;
+        }
+
+        return false;
+    }
+    
+    public override IInteraction[] GenerateInteractions(InteractionEvent interactionEvent)
+    {
+        List<IInteraction> list = base.GenerateInteractions(interactionEvent).ToList();
+     
+        // Adds the interaction if it's not playing
+        if (CanConsume())
+        {
+            list.Add(new ConsumeInteraction {icon = sprite, verb = consumeVerb});
+        }
```

Comment:
Conditions for interactions should only be checked inside of them, as they could be executed at a later time, bypassing this check.